<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServerPanel</class>
 <widget class="QWidget" name="ServerPanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>5</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <property name="spacing">
    <number>10</number>
   </property>
   <item>
    <widget class="QSplitter" name="mainSplitter">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <!-- Right Panel -->
     <widget class="QWidget" name="rightPanel">
      <layout class="QVBoxLayout" name="rightLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTabWidget" name="dataTabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="treeTab">
          <attribute name="title">
           <string>数据模型</string>
          </attribute>
          <layout class="QVBoxLayout" name="treeTabLayout">
           <!-- DataTreeWidget will be inserted here -->
          </layout>
         </widget>
         <widget class="QWidget" name="datasetTab">
          <attribute name="title">
           <string>数据集</string>
          </attribute>
          <layout class="QVBoxLayout" name="datasetTabLayout">
           <item>
            <widget class="QSplitter" name="datasetSplitter">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <!-- Left Panel: DataSet List + Control Blocks -->
             <widget class="QWidget" name="datasetLeftPanel">
              <layout class="QVBoxLayout" name="datasetLeftLayout">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QSplitter" name="leftVerticalSplitter">
                 <property name="orientation">
                  <enum>Qt::Orientation::Vertical</enum>
                 </property>
                 <!-- Top: DataSet List -->
                 <widget class="QWidget" name="datasetListPanel">
                  <layout class="QVBoxLayout" name="datasetListLayout">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="datasetListLabel">
                     <property name="styleSheet">
                      <string notr="true">font-weight: bold; padding: 5px;</string>
                     </property>
                     <property name="text">
                      <string>数据集列表</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QListWidget" name="datasetList">
                     <property name="alternatingRowColors">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                 <!-- Bottom: Control Blocks -->
                 <widget class="QWidget" name="controlBlockPanel">
                  <layout class="QVBoxLayout" name="controlBlockPanelLayout">
                   <property name="leftMargin">
                    <number>0</number>
                   </property>
                   <property name="topMargin">
                    <number>0</number>
                   </property>
                   <property name="rightMargin">
                    <number>0</number>
                   </property>
                   <property name="bottomMargin">
                    <number>0</number>
                   </property>
                   <item>
                    <widget class="QGroupBox" name="controlBlockGroup">
                     <property name="title">
                      <string>关联的控制块</string>
                     </property>
                     <layout class="QVBoxLayout" name="controlBlockLayout">
                      <item>
                       <widget class="QTableWidget" name="controlBlockTable">
                        <property name="alternatingRowColors">
                         <bool>true</bool>
                        </property>
                        <property name="selectionBehavior">
                         <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
                        </property>
                        <property name="columnCount">
                         <number>2</number>
                        </property>
                        <column>
                         <property name="text">
                          <string>属性</string>
                         </property>
                        </column>
                        <column>
                         <property name="text">
                          <string>值</string>
                         </property>
                        </column>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </widget>
               </item>
              </layout>
             </widget>
             <!-- Right Panel: FCDA List -->
             <widget class="QWidget" name="fcdaPanel">
              <layout class="QVBoxLayout" name="fcdaPanelLayout">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="datasetNameLabel">
                 <property name="styleSheet">
                  <string notr="true">font-weight: bold; font-size: 13px; padding: 5px;</string>
                 </property>
                 <property name="text">
                  <string>选择数据集以查看详情</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QGroupBox" name="fcdaGroup">
                 <property name="title">
                  <string>FCDA 列表</string>
                 </property>
                 <layout class="QVBoxLayout" name="fcdaLayout">
                  <item>
                   <widget class="QTableWidget" name="fcdaTable">
                    <property name="alternatingRowColors">
                     <bool>true</bool>
                    </property>
                    <property name="selectionBehavior">
                     <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
                    </property>
                    <property name="columnCount">
                     <number>6</number>
                    </property>
                    <column>
                     <property name="text">
                      <string>逻辑设备</string>
                     </property>
                    </column>
                    <column>
                     <property name="text">
                      <string>逻辑节点</string>
                     </property>
                    </column>
                    <column>
                     <property name="text">
                      <string>DO</string>
                     </property>
                    </column>
                    <column>
                     <property name="text">
                      <string>DA</string>
                     </property>
                    </column>
                    <column>
                     <property name="text">
                      <string>功能约束</string>
                     </property>
                    </column>
                     <column>
                     <property name="text">
                      <string>值</string>
                     </property>
                    </column>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        <widget class="QWidget" name="connectionTab">
         <attribute name="title">
          <string>客户端连接</string>
         </attribute>
         <layout class="QVBoxLayout" name="connectionTabLayout">
          <item>
           <widget class="QGroupBox" name="connectionGroup">
            <property name="title">
             <string>客户端连接</string>
            </property>
            <layout class="QVBoxLayout" name="connectionLayout">
             <item>
              <widget class="QTableWidget" name="connectionTable">
               <property name="alternatingRowColors">
            <bool>true</bool>
               </property>
               <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
               </property>
               <property name="columnCount">
            <number>3</number>
               </property>
               <column>
            <property name="text">
             <string>客户端IP</string>
            </property>
               </column>
               <column>
            <property name="text">
             <string>连接时间</string>
            </property>
               </column>
               <column>
            <property name="text">
             <string>状态</string>
            </property>
               </column>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
         <widget class="QWidget" name="simulationTab">
          <attribute name="title">
           <string>仿真设置</string>
          </attribute>
          <layout class="QVBoxLayout" name="simulationTabLayout">
           <item>
            <widget class="QGroupBox" name="simConfigGroup">
             <property name="title">
              <string>仿真配置</string>
             </property>
             <layout class="QGridLayout" name="simConfigLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="simModeLabel">
                <property name="text">
                 <string>仿真模式:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QComboBox" name="simModeCombo">
                <item>
                 <property name="text">
                  <string>随机值</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>正弦波</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>阶跃</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>自定义脚本</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="simIntervalLabel">
                <property name="text">
                 <string>更新间隔(ms):</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSpinBox" name="simIntervalInput">
                <property name="minimum">
                 <number>100</number>
                </property>
                <property name="maximum">
                 <number>10000</number>
                </property>
                <property name="singleStep">
                 <number>100</number>
                </property>
                <property name="value">
                 <number>1000</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="simScriptGroup">
             <property name="title">
              <string>仿真脚本</string>
             </property>
             <layout class="QVBoxLayout" name="simScriptLayout">
              <item>
               <widget class="QTextEdit" name="simScriptEdit">
                <property name="placeholderText">
                 <string># Python仿真脚本
# 使用 set_value(reference, value) 设置数据点值
# 使用 get_value(reference) 获取数据点值</string>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="simScriptBtnLayout">
                <item>
                 <widget class="QPushButton" name="runScriptBtn">
                  <property name="text">
                   <string>▶ 运行脚本</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="stopScriptBtn">
                  <property name="text">
                   <string>⏹ 停止</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="simScriptSpacer">
                  <property name="orientation">
                   <enum>Qt::Orientation::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer name="simVerticalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
